### DISCUSSION TITLE
Help with clink lua format

### DISCUSSION THREAD

**Discussion by N6REJ** [chrisant996/clink#341] [Q&A] [answered]
We took over a project and the .lua for clink has us confused.
We'd like to be able to get rid of the dark red color for the root folder and set the default font and can't seem to understand where/how this is happening.  We see things like this...
```
local console_prompt = "\x1b[1;35;40m{username}@{computername} \x1b[1;32;40m{cwd} {git}{hg} \n\x1b[1;30;40m{dollar} \x1b[0m"
```
Our repo is at 
https://github.com/Bearsampp/module-consolez/tree/main/bin/consolez1.19.0.19104

Any help will be greatly appreciated.

**Comment by chrisant996** [comment, marked as answer]
I'm not sure if you're asking what [ANSI escape codes](https://en.wikipedia.org/wiki/ANSI_escape_code) are, or if you're asking what `\x1b` means in Lua strings.

`\x1b` = hex 1b = decimal 27 = ESC, which is the ANSI escape character.

`\x1b[ ... m` is an escape code for Select Graphic Rendition, which controls colors and bold and underline and reverse video.

In the code `\x1b[1;32;40m`, the `1` means "bold" (i.e. bright), `32` means "red foreground", and `40` means "black background".

Refer to [ANSI escape codes](https://en.wikipedia.org/wiki/ANSI_escape_code) for more info about escape codes.

**Reply by N6REJ** [reply]
your a godsend!!!  tyvm!
